src/theme/DocRoot/Layout/Sidebar/index.tsx

```diff
@@ -1,51 +1,74 @@
 import React, {useEffect, useState} from "react"
+import {useHistory} from "react-router-dom"
 import Sidebar from "@theme-original/DocRoot/Layout/Sidebar"
 import Search from "docusaurus-lunr-search/src/theme/SearchBar"
-import {useHistory} from "react-router-dom"
 import useIsBrowser from "@docusaurus/useIsBrowser"
+
 import PageSearchIcon from "@site/static/icons/basic/page-search.svg"
 import EnterKeyIcon from "@site/static/icons/basic/enter-key.svg"
 import UpDownKeyIcon from "@site/static/icons/basic/up-down-key.svg"
 import EscapeKeyIcon from "@site/static/icons/basic/escape-key.svg"
-import styles from "./Sidebar.module.css"
+import styles from "./styles.module.css"
 
-export default function SidebarWrapper(props) {
-  const [isSearchModalVisible, setIsSearchModalVisible] = useState(false)
+const CustomSearch = () => {
+  const [isSearchModalVisible, setIsSearchModalVisible] = useState<boolean>(false)
   const history = useHistory()
   const isBrowser = useIsBrowser()
-  const placeholder = isBrowser && window.navigator.platform.startsWith("Mac") ? "Search ⌘+K" : "Search Ctrl+K"
+  const placeholder = isBrowser
+    ? window.navigator.userAgent.startsWith("Mac")
+      ? "Search ⌘+K"
+      : "Search Ctrl+K"
+    : "Search"
 
+  // Function to handle opening the search modal
   function handleSearchClick() {
     setIsSearchModalVisible(true)
   }
 
+  // Function to handle closing the search modal
   function handleSearchModalClose() {
     setIsSearchModalVisible(false)
   }
 
+  // Function to control body scroll based on modal visibility
   function setBodyScroll() {
     document.body.style.overflow = isSearchModalVisible ? "hidden" : "auto"
   }
 
+  // Effect to handle body scroll based on modal visibility changes
   useEffect(() => {
     setBodyScroll()
     return () => {
       document.body.style.overflow = "initial"
     }
   }, [isSearchModalVisible])
 
-  function handleKeyPress(event) {
+  // Effect to focus on search input when modal becomes visible
+  useEffect(() => {
+    if (isSearchModalVisible) {
+      setTimeout(() => {
+        const searchInput = document.getElementById("search_input_react")
```

Comment:
AFAIK https://react.dev/reference/react/useRef should be used for scenarios like this

Reply:
The Search component from lunar search does not allows us to attach a ref to it, so to handle the autofocus when search is open on desktop I am accessing the search rendered by lunar search by Id and focusing it.